flycheck.el

```diff
@@ -6972,6 +6973,59 @@ See URL `http://stylelint.io/'."
   :error-parser flycheck-parse-stylelint
   :modes (css-mode))
 
+(flycheck-def-args-var flycheck-cuda-args cuda
+  :package-version '(flycheck . "0.01"))
+
+(flycheck-def-option-var flycheck-cuda-language-standard nil cuda
+  "Our CUDA Language Standard "
+  :type '(choice (const :tag "Default standard" nil)
+		 (string :tag "Language standard"))
+  :safe #'stringp
+  :package-version '(flycheck . "0.01"))
+(make-variable-buffer-local 'flycheck-cuda-language-standard)
+
+(flycheck-def-option-var flycheck-cuda-includes nil cuda
+  "Our include directories to pass to nvcc "
+  :type '(repeat (file :tag "Include file"))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.01"))
+
+(flycheck-def-option-var flycheck-cuda-definitions nil cuda
+  "Additional preprocessor definitions for nvcc.
+A list of strings to pass to cuda, a la flycheck-clang"
+  :type '(repeat (string :tag "Definitions"))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.01"))
+
+(flycheck-def-option-var flycheck-cuda-include-path nil cuda
+  "A list of include directories for nvcc."
+  :type '(repeat (directory :tag "Include directory"))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.01"))
+
+(flycheck-define-checker cuda
+  "A CUDA C/C++ syntax checker..
+
+See URL `https://developer.nvidia.com/cuda-llvm-compiler'."
+  :command ("nvcc"
+            "-c" ;; Compile Only
+            (option "-std=" flycheck-cuda-language-standard concat)
+            (option-list "-include" flycheck-cuda-includes)
+            (option-list "-D" flycheck-cuda-definitions concat)
+            (option-list "-I" flycheck-cuda-include-path)
+            ;; Read from standard input
+            source)
```

Comment:
`source` is not standard input: look at the docstring for `flycheck-substitute-argument`.  If you want standard input, you specify that explicitly using `:standard-input t`, and usually pass a special argument to the checker (like `-`).

So either use `source` and remove the comment, or look to see if `nvcc` supports standard input and use that (we prefer standard input if available).

Reply:
I've removed the comment and kept it as source for now. 

I'll probably add more bells and whistles to this as I get more comfortable with contributing.